src/p2pool.cpp

```diff
@@ -391,6 +391,31 @@ void p2pool::handle_chain_main(ChainMain& data, const char* extra)
 				LOGINFO(0, log::LightCyan() << "You received a payout of " << log::LightGreen() << log::XMRAmount(payout) << log::LightCyan() << " in block " << log::LightGreen() << data.height);
 			}
 			api_update_block_found(&data);
+
+			if(!params().m_onBlockFound.empty())
+			{
+			    std::stringstream cmd;
+				double po = payout;
+				if(po > 0)
+				{
+					po = po * 0.000000000001; //convert to XMR reward
+				}
+
+			    cmd << params().m_onBlockFound << " BLOCK " << data.id << " " << data.timestamp << " " << data.reward << " " << po;
+
+			    //system() requires char and not string, convert to char
+			    std::string ss(cmd.str());
+
+                char sys[ss.length() + 1];
+                strcpy(sys, ss.c_str());
+
+                //If there is an error, then log it
+                if(!system(sys))
```

Comment:
This is not cross-platform compatible. You should have a function in util.cpp that does it and add all platform #ifdefs there.

Reply:
Also, this function waits for the process to finish which can block p2pool's thread. Can you use some alternative that doesn't wait?

Reply:
I'm working on this. I'm not a c++ programer but I'll figure it out.